streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -791,12 +788,16 @@ public boolean conditionMet() {
 
         final ReadOnlyKeyValueStore<String, String> store2 = kafkaStreams.store(storeName, QueryableStoreTypes.<String, String>keyValueStore());
 
-        TestUtils.waitForCondition(new TestCondition() {
-            @Override
-            public boolean conditionMet() {
-                return "125".equals(store2.get("a")) && "34".equals(store2.get("b"));
-            }
-        }, maxWaitMs, "wait for agg to be '123'");
+        try {
+            TestUtils.waitForCondition(new TestCondition() {
+                @Override
+                public boolean conditionMet() {
+                    return store2.get("a").endsWith("125") && store2.get("b").endsWith("34");
+                }
+            }, maxWaitMs, "wait for agg to be <a,125> and <b,34>");
```

Comment:
Chatted with @mjsax , I think the root cause is that wthout eos, we do not wipe out the state store at all even if there is no checkpoint files written locally and always re-process the data sine the last committed offset of from reset policy specified offset, hence at-least-once.

I'm wondering though, that in a rare condition we could commit after "1" is written, and failed after "1+2" is written, so is it possible that the end state will be "1225"?